### ISSUE TITLE
Print git hash in version output and add -version

### ISSUE THREAD

**Issue by Timmmm** [rems-project/sail#204]
Hi, I've been recompiling a Sail a bit and finding Opam *very* confusing. It would be really helpful if `sail` had a `sail -version` option that printed the version *and git hash* so I know if Opam has actually built the version I want. Also add `-version` as an alias to `-v` because I only just realised that there even was a version option.

(Also make sure it doesn't *require* the source code to be a `git` repo - I've seen projects do that before and it's quite annoying because then you can't build from source tarballs.)

**Comment by Alasdair** [comment]
Currently `sail -v` includes the git commit number when not built using opam, i.e. when using `make install` (which just uses dune). If you are recompiling a lot, you can have opam only manage dependencies using `opam install . --deps-only` then build manually.

**Comment by Timmmm** [comment]
Ah ok, sounds like I just need to read how to use `opam`!

**Comment by Timmmm** [comment]
FWIW I don't think the git hash in the output is quite right - you can `make install` sail, then checkout a different commit, `make install` again and it will still show the old commit. `make clean; make install` updates it to the new one, so something is missing in the dependency graph.

Also I'm going to reopen this because I *keep* trying to run `sail --version` and getting shouted at! :-)

**Comment by Alasdair** [comment]
I hadn't noticed this before, but it seems dune doesn't pick up that things have changed after a `git checkout`. Maybe we could add `dune clean` as a post-commit hook, but not sure how well that would work.

**Comment by Alasdair** [comment]
Or it might be that the rule that generates the file containing the commit needs to run every build

**Comment by Alasdair** [comment]
Yes, the problem is that after `git checkout` the `manifest.ml` from the previous checkout still exists, so the build rule:
```
(rule
  (target manifest.ml)
  (mode fallback)
  (action
    (with-outputs-to %{target}
      (chdir %{workspace_root}
        (run sail_manifest -gen_manifest)))))
```
doesn't fire

I think everything else should be fine, just the version info will be incorrect until a `dune clean` is run.
